### ISSUE TITLE
Support premade schema

### ISSUE THREAD

**Issue by stevefan1999** [nestjs/graphql#49]
## I'm submitting a...

<pre><code>
[ ] Regression 
[ ] Bug report
[x] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

`typePaths` is mandatory and dominant, without it on graphql.mergeTypes it will throw an error.

## Expected behavior

I should be able to use a pre-cooked schema out of the box.

## Minimal reproduction of the problem with instructions

```ts
    GraphQLModule.forRootAsync({
      imports: [
        TypeGQLModule.forSchema({
          resolvers: [
            DefaultResolver,
            ...ModuleLocator.flattenModuleField('resolvers')
          ],
          pubSub,
          authChecker
        })
      ],
      async useFactory(graphQL: GraphQlBridge): Promise<GqlModuleOptions> {
        const schema: GraphQLSchema = graphQL.buildSchema()

        const playground: any = {
          settings: {
            'editor.cursorShape': 'line'
          }
        }

        return {
          schema,
          introspection: true,
          tracing: true,
          context: ({ req, res }) => ({
            req,
            res
          }),
          playground
        }
      },
      inject: [GraphQlBridge]
    })
  ],
```
It failed with:
```ts
Error: Specified query type "Query" not found in document.
    at E:\typescript-starter\node_modules\graphql\utilities\buildASTSchema.js:184:15
    at Array.forEach (<anonymous>)
    at getOperationTypes (E:\typescript-starter\node_modules\graphql\utilities\buildASTSchema.js:177:27)
    at Object.buildASTSchema (E:\typescript-starter\node_modules\graphql\utilities\buildASTSchema.js:127:36)
    at Object.buildSchemaFromTypeDefinitions (E:\typescript-starter\node_modules\graphql-tools\dist\generate\buildSchemaFromTypeDefinitions.js:24:28)
    at Object.makeExecutableSchema (E:\typescript-starter\node_modules\graphql-tools\dist\makeExecutableSchema.js:27:29)
    at GraphQLFactory.mergeOptions (E:\typescript-starter\node_modules\@nestjs\graphql\dist\graphql.factory.js:30:98)
    at Function.<anonymous> (E:\typescript-starter\node_modules\@nestjs\graphql\dist\graphql.module.js:73:55)
    at Generator.next (<anonymous>)
    at E:\typescript-starter\node_modules\@nestjs\graphql\dist\graphql.module.js:19:71
    at new Promise (<anonymous>)
    at __awaiter (E:\typescript-starter\node_modules\@nestjs\graphql\dist\graphql.module.js:15:12)
    at Object.useFactory [as metatype] (E:\typescript-starter\node_modules\@nestjs\graphql\dist\graphql.module.js:71:68)
    at resolveConstructorParams (E:\typescript-starter\node_modules\@nestjs\core\injector\injector.js:68:55)
    at Injector.resolveConstructorParams (E:\typescript-starter\node_modules\@nestjs\core\injector\injector.js:99:30)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```

## What is the motivation / use case for changing the behavior?
I used [MagnusCloudCorp/nestjs-type-graphql](https://github.com/MagnusCloudCorp/nestjs-type-graphql) instead of the helpers from `@nestjs/graphql` provided out of the box and TypeGraphQL provided a compiled schema instead of SDL.

## Environment

<pre><code>
Nest version: 5.1.0

</code></pre>

## Extra info
This is the reason it failed:
https://github.com/nestjs/graphql/blob/b58d3e50dc9672a1047ef192587849fbbfc42281/lib/graphql.factory.ts#L22-L37
My schema option, no matter what are always gonna be `Object.assign`'d

**Comment by stevefan1999** [comment]
I think this is the culprit🤔
https://github.com/nestjs/graphql/blob/b58d3e50dc9672a1047ef192587849fbbfc42281/lib/graphql.factory.ts#L44-L46

**Comment by kamilmysliwiec** [comment]
What exactly would you need to make this working correctly together? The possibility to pass a schema created beforehand?

**Comment by stevefan1999** [comment]
@kamilmysliwiec Exactly! I tried to implement a quick fix with `mergeSchemas`.

**Comment by kamilmysliwiec** [comment]
Just merged and published as 5.1.1
